lib/hermann/producer.rb

```diff
@@ -62,7 +62,7 @@ def push(value, opts={})
       end
 
       if Hermann.jruby?
-        result = @internal.push_single(value, topic, nil)
+        result = @internal.push_single(value, topic, opts.fetch(:key, ""))
```

Comment:
Yes, let me explicitly pass nil instead, which is what happens when you pass no partition key in directly